sources/osgUtil/DisplayNormalVisitor.js

```diff
@@ -51,7 +51,7 @@ CompilerOffsetNormal.prototype = MACROUTILS.objectInherit( Compiler.prototype, {
         vertexOffset = this.createVariable( 'vec3', 'vertexOffset' );
 
         // normalize len (divide scale of worldmat, don't work for non uniform scale, but work enough for debug)
-        var str = '%out = %offset == 1.0 ? %vertex + normalize(%vecOffset.xyz) * %scale / %world[0][0] : %vertex;';
+        var str = '%out = %offset == 1.0 ? %vertex + normalize(%vecOffset.xyz) * %scale / length(%world[0].xyz) : %vertex;';
```

Comment:
Yes, it's a direction (not normalized though)

getOrCreateVertexAttribute return the vertex in local space (morphed and not rigged), we can't simply return a world pos (need to hook another function)
+
we want to "normalize" the length of the debugged normal to be consistent in screen space.
We don't want it to be worldunit = 1